rsconnect/main.py

```diff
@@ -1038,45 +963,141 @@ def deploy_html(
+    @deploy.command(
+        name=alias,
+        short_help="Deploy a {desc} to RStudio Connect [v{version}+].".format(
+            desc=app_mode.desc(), version=min_version
+        ),
+        help=(
+            'Deploy a {desc} module to RStudio Connect. The "directory" argument must refer to an '
+            "existing directory that contains the application code."
+        ).format(desc=app_mode.desc()),
+    )
+    @server_args
+    @content_args
+    @click.option(
+        "--entrypoint",
+        "-e",
+        help=(
+            "The module and executable object which serves as the entry point for the {desc} (defaults to app)"
+        ).format(desc=app_mode.desc()),
+    )
+    @click.option(
+        "--exclude",
+        "-x",
+        multiple=True,
+        help=(
+            "Specify a glob pattern for ignoring files when building the bundle. Note that your shell may try "
+            "to expand this which will not do what you expect. Generally, it's safest to quote the pattern. "
+            "This option may be repeated."
+        ),
+    )
+    @click.option(
+        "--python",
+        "-p",
+        type=click.Path(exists=True),
+        help=(
+            "Path to Python interpreter whose environment should be used. "
+            "The Python environment must have the rsconnect package installed."
+        ),
+    )
+    @click.option(
+        "--conda",
+        "-C",
+        is_flag=True,
+        hidden=True,
+        help="Use Conda to deploy (requires Connect version 1.8.2 or later)",
+    )
+    @click.option(
+        "--force-generate",
+        "-g",
+        is_flag=True,
+        help='Force generating "requirements.txt", even if it already exists.',
+    )
+    @click.option(
+        "--image",
+        "-I",
+        help="Target image to be used during content execution (only applicable if the RStudio Connect "
+        "server is configured to use off-host execution)",
+    )
+    @click.argument("directory", type=click.Path(exists=True, dir_okay=True, file_okay=False))
+    @click.argument(
+        "extra_files",
+        nargs=-1,
+        type=click.Path(exists=True, dir_okay=False, file_okay=True),
+    )
+    def deploy_app(
+        name,
+        server,
+        api_key,
+        insecure,
+        cacert,
+        entrypoint,
+        exclude,
+        new,
         app_id,
-        app_mode,
-        deployment_name,
         title,
-        default_title,
-        bundle,
+        python,
+        conda,
+        force_generate,
+        verbose,
+        directory,
+        extra_files,
         env_vars,
-    )
+        image,
+    ):
+        kwargs = locals()
+        entrypoint = validate_entry_point(entrypoint, directory)
+        environment = create_python_environment(
+            directory,
+            extra_files,
+            force_generate,
+            python,
+            conda,
+        )
+
+        rsce = api.RSConnectExecutor(**kwargs)
+        (
```

Comment:
The parentheses and indentation aren't needed here since you're including `rsce` on the line below.

Reply:
The parentheses are for omitting line breaks. Removing them will mean every line will have to add '\'